classes/order/OrderHistory.php

```diff
@@ -99,6 +99,8 @@ public function changeIdOrderState($new_order_state, $id_order, $use_existing_pa
         // executes hook
         if (in_array($new_os->id, array(Configuration::get('PS_OS_PAYMENT'), Configuration::get('PS_OS_WS_PAYMENT')))) {
             Hook::exec('actionPaymentConfirmation', array('id_order' => (int)$order->id), null, false, true, false, $order->id_shop);
+            //initialize invoice date
+            $order->invoice_date =date('Y-m-d H:i:s');
```

Comment:
Hi,

I disagree with that solution. Even if there is no invoice, it would be accounted as invoiced because there is a invoice_date. And the invoice will appear in the next status, so the invoice date will be wrong.

And legally, if we set "invoice date" whereas there is no invoice, merchants will have problems.